dask/dataframe/groupby.py

```diff
@@ -280,6 +281,40 @@ def _groupby_get_group(df, by_key, get_key, columns):
         return df.iloc[0:0]
 
 
+def numeric_only_deprecate_default(func):
+    """Decorator for methods that should warn when numeric_only is default"""
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        numeric_only = kwargs.get("numeric_only", no_default)
+        if PANDAS_GT_150 and not PANDAS_GT_200 and numeric_only is no_default:
+            warnings.warn(
+                "The default value of numeric_only will be `False` "
+                "in a future version of Pandas.",
```

Comment:
I think we can be more specific about when this change will occur. What do you think about something like? 

> The default value of numeric_only will be `False` when using dask with pandas 2.0

Reply:
Also, this warning is getting elevated to an error in CI and causing a bunch of tests to fail. I'd recommend using `with pytest.warns(...):` to catch / ensure this warning is getting emitted in the places we expect it to